src/iris/bin/sync_targets.py

```diff
@@ -444,7 +444,12 @@ def sync_ldap_lists(ldap_settings, engine):
     user_add_count = 0
 
     for list_cn, list_name in ldap_lists:
-        members = get_ldap_flat_membership(l, ldap_settings['search_strings'], list_cn, ldap_settings['max_depth'], 0, set())
+        try:
+            members = get_ldap_flat_membership(l, ldap_settings['search_strings'], list_cn, ldap_settings['max_depth'], 0, set())
+        except ldap.SERVER_DOWN:
+            # reconnect and retry once
+            l.reconnect(ldap_settings['connection']['url'])
+            members = get_ldap_flat_membership(l, ldap_settings['search_strings'], list_cn, ldap_settings['max_depth'], 0, set())
```

Comment:
Can we log saying we got disconnected and then reconnected? Adding a metric for this might also be useful.